modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APISchemaRelevantObjectEntryModelListener.java

```diff
@@ -107,6 +108,40 @@ private void _validate(ObjectEntry objectEntry) {
 		}
 	}
 
+	private void _validateExternalReferenceCode(
+		ObjectEntry objectEntry, ObjectEntry originalObjectEntry) {
+
+		try {
+			Map<String, Serializable> originalObjectEntryValues =
+				originalObjectEntry.getValues();
+
+			if (Validator.isNotNull(
+					_objectEntryHelper.getObjectEntry(
+						objectEntry.getCompanyId(),
+						StringBundler.concat(
+							"externalReferenceCode eq '",
+							objectEntry.getExternalReferenceCode(),
+							"' and r_apiApplicationToAPISchemas_c_",
+							"apiApplicationId ne '",
+							originalObjectEntryValues.get(
+								"r_apiApplicationToAPISchemas_c_" +
+									"apiApplicationId"),
+							"'"),
+						"L_API_SCHEMA"))) {
+
+				throw new ObjectEntryValuesException.InvalidObjectField(
```

Comment:
I dont know if we should return "The external reference code is already in use", which is the message returned by the objects layer. Just to mantain consistency. It's just a question for debating, I'm not sure about it...